### PR TITLE
api to handle all post fetch

### PR DESCRIPTION
**Pull request by warrenhhh**

### CODE REVIEW

File: Backend/marketuserpost.js [hunk 1]
```diff
@@ -54,6 +54,30 @@ async function marketcreatePost(req, res) {
     }
 }
 
+
+/**
+ * Function to retrieve market posts and sort them by createdAt 
+ */
```
Comment (balamurali0532) [inline Backend/marketuserpost.js:61]: missing comments

File: Backend/marketuserpost.js [hunk 2]
```diff
@@ -54,6 +54,30 @@ async function marketcreatePost(req, res) {
...
+async function retrieveMarketPosts(req, res) {
+    try {
+        const marketPosts = await Marketpost.find().sort({ createdAt: 'desc' });
+        const marketPostIterator = new Iterator(marketPosts);
+        const result = [];
+        let marketPost = marketPostIterator.next();
+        while (!marketPost.done) {
+            result.push(marketPost.value);
+            console.log(marketPost);
```
Comment (balamurali0532) [inline Backend/marketuserpost.js]: remove the console.log

File: Backend/marketuserpost.js [hunk 3]
```diff
@@ -54,6 +54,30 @@ async function marketcreatePost(req, res) {
     }
 }
 
+
+/**
+ * Function to retrieve market posts and sort them by createdAt 
+ */
+async function retrieveMarketPosts(req, res) {
+    try {
+        const marketPosts = await Marketpost.find().sort({ createdAt: 'desc' });
```
Comment (balamurali0532) [inline Backend/marketuserpost.js]: better to break it into parts instead of keep await the mongo query

File: Backend/marketuserpost.js (hunk 1 above)
Comment (warrenhhh) [reply Backend/marketuserpost.js:61]: code is commented.

File: Backend/marketuserpost.js (hunk 2 above)
Comment (warrenhhh) [reply Backend/marketuserpost.js]: console.log is removed.

File: Backend/marketuserpost.js (hunk 3 above)
Comment (warrenhhh) [reply Backend/marketuserpost.js]: the function is divided.

**Review by balamurali0532 — APPROVED** [review]
review is done

**Pull request merged**